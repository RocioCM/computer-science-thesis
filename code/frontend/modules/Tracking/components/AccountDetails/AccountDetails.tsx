import { useEffect, useState } from 'react';
import useModal from '@/common/hooks/useModal';
import TrackingServices from '../../services';
import { UserData } from '../../types';
import LoadingSpinner from '@/common/components/LoadingSpinner';

interface Props {
  blockchainId: string;
}

const AccountDetails: React.FC<Props> = ({ blockchainId }) => {
  const [userData, setUserData] = useState<UserData | null>(null);
  const { Modal, showModal } = useModal();

  const fetchUserData = async (blockchainId: string) => {
    setUserData(null);
    const { ok, data } = await TrackingServices.getUserPublicData(blockchainId);
    if (ok) {
      setUserData(data);
    }
  };

  useEffect(() => {
    if (blockchainId) fetchUserData(blockchainId);
  }, [blockchainId]);

  return (
    <>
      <span
        className={userData ? 'underline cursor-pointer' : ''}
        onClick={userData ? showModal : undefined}
      >
        {userData?.userName ? userData.userName : blockchainId}
      </span>
      <Modal>
        {userData ? (
          <div className="w-full my-auto space-y-xs">
            <p>
              <span className="font-semibold">Nombre:</span> {userData.userName}
            </p>
            <p>
              <span className="font-semibold">Email:</span> {userData.email}
            </p>
            <p>
              <span className="font-semibold">Tel√©fono:</span> {userData.phone}
            </p>
            <p className="w-full whitespace-preline">
              <span className="font-semibold">Blockchain ID:</span>{' '}
              {userData.blockchainId}
            </p>
          </div>
        ) : (
          <div className="flex-1 flex items-center">
            <LoadingSpinner />
          </div>
        )}
      </Modal>
    </>
  );
};

export default AccountDetails;
