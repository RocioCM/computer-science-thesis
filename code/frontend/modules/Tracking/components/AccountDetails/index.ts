import AccountDetails from './AccountDetails';

export default AccountDetails;
