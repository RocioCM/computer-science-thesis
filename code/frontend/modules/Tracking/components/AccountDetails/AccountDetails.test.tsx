import { render, screen, fireEvent, act } from '@testing-library/react';
import AccountDetails from './AccountDetails';

describe('Tracking - AccountDetails', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  });

  afterEach(() => {
    const modalRoot = document.getElementById('modal-root');
    if (modalRoot) {
      document.body.removeChild(modalRoot);
    }
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <AccountDetails blockchainId={'0x123456789'} />
    );
    expect(container).toMatchSnapshot();
  });
});
