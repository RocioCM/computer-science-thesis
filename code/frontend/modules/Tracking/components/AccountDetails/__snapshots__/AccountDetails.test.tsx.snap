// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracking - AccountDetails Matches snapshot 1`] = `
<div>
  <span
    class=""
  >
    0x123456789
  </span>
</div>
`;
